### DISCUSSION TITLE
WebView not secure error

### DISCUSSION THREAD

**Discussion by PrakharDoneria** [Sketchware-Pro/Sketchware-Pro#703] [Q&A] [answered]
I am making a app based on WebView but when I try logging Google it says the browser (WebView) is not safe also the YouTube player doesn't work with a error that WebView is too old. I am using sketchware latest version v6.4.0 rc-05-minAPI26
For more information refer stack overflow: https://stackoverflow.com/questions/75086618/insecure-browser-error-in-webview-on-logging-google

**Comment by waelchateur** [comment]
> I am making a app based on WebView but when I try logging Google it says the browser (WebView) is not safe also the YouTube player doesn't work with a error that WebView is too old. I am using sketchware latest version v6.4.0 rc-05-minAPI26 For more information refer stack overflow: https://stackoverflow.com/questions/75086618/insecure-browser-error-in-webview-on-logging-google

i suggest you to change the user agent or you can simply use newer lib
androidx.webkit:webkit:1.6.0-rc01
dont add webview widget just add simple linear and convert it to webview

**Comment by j-px** [comment, marked as answer]
It's getting blocked due to this: https://developers.googleblog.com/2021/06/upcoming-security-changes-to-googles-oauth-2.0-authorization-endpoint.html

> Embedded webview libraries are problematic because they allow a nefarious developer to intercept and alter communications between Google and its users by acting as a "man in the middle." An application embedding a webview can modify or intercept network requests, insert custom scripts that can potentially record every keystroke entered in a login form, access session cookies, or alter the content of the webpage.

Summary: Don't make users sign in with their Google account in your app's WebView. Open the default browser instead, if you really want them to.

**Reply by PrakharDoneria** [reply]
But signup is working in my old app that i made last month 
![Screenshot_2023-01-12_103156](https://user-images.githubusercontent.com/92568876/211981632-bf77be40-0134-4150-9463-278ca63e9f29.png)

**Reply by j-px** [reply]
You can change the WebView's user agent so that signing in doesn't get blocked, but you should know that the block has a reason behind it.
